tools/restatectl/src/commands/partition/list.rs

```diff
@@ -79,27 +107,85 @@ pub async fn list_partitions(
                     let host_node =
                         GenerationalNodeId::new(host.id, host.generation.expect("generation"));
                     let details = PartitionListEntry { host_node, status };
-                    partitions.entry(partition_id).or_default().push(details);
+                    partitions.push((partition_id, details));
                 }
             }
         }
     }
+
     // Show information organized by partition and node
     let mut partitions_table = Table::new_styled();
     partitions_table.set_styled_header(vec![
         "P-ID",
         "NODE",
         "MODE",
         "STATUS",
-        "APPLIED",
-        "PERSISTED",
         "LEADER",
         "EPOCH",
+        "SEQUENCER",
+        "APPLIED",
+        "PERSISTED",
         "SKIPPED",
         "LAST-UPDATE",
     ]);
-    for (partition_id, processors) in partitions {
-        for processor in processors {
+
+    partitions
+        .into_iter()
+        .sorted_by(|a, b| match opts.sort {
+            SortMode::Partition => a.0.cmp(&b.0),
+            SortMode::Node => {
+                a.1.host_node
+                    .cmp(&b.1.host_node)
+                    .then_with(|| a.0.cmp(&b.0))
+            }
+            SortMode::Active => {
+                a.1.status
+                    .effective_mode
+                    .cmp(&b.1.status.effective_mode)
+                    .then_with(|| a.1.host_node.cmp(&b.1.host_node))
+            }
+        })
+        .for_each(|(partition_id, processor)| {
+            let is_leader = processor
+                .status
+                .last_observed_leader_node
+                .map(|n| {
+                    n.generation.is_some_and(|g| {
+                        PlainNodeId::from(n.id).with_generation(g) == processor.host_node
+                    })
+                })
+                .unwrap_or_default();
+
+            let (in_tail_segment, maybe_sequencer) = logs
+                .get(&LogId::from(partition_id))
+                .map(|chain| {
+                    let tail = chain.tail();
+                    let in_tail = processor
+                        .status
+                        .last_applied_log_lsn
+                        .map(Lsn::from)
+                        .is_some_and(|applied_lsn| applied_lsn.ge(&tail.base_lsn));
```

Comment:
So in catch-up mode, we are displaying nothing? What is the meaning of the `sequencer` column then?

Reply:
It's in the PR description but I've now added a longer help message about it too, it shows the loglet's sequencer, for rows displaying the leader partition processor.